File: Data Structures & Algorithms/sliding-window-maximum/submission-1.cs
public class Solution {
    public int[] MaxSlidingWindow(int[] nums, int k) {
        int n = nums.Length;
        var res = new int[n - k + 1];

        for (int i = 0; i <= n - k; i++) {
            int max = nums[i];
            for (int j = i; j < i + k; j++) {
                max = Math.Max(max, nums[j]);
            }
            res[i] = max;
        }

        return res;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.ts
class Solution {
    /**
     * @param {string} s
     * @return {number}
     */
    lengthOfLongestSubstring(s: string): number {
        const set = new Set();
        let l = 0;
        let res = 0;
        for (let r = 0; r < s.length; r++) {
            while (set.has(s[r])) {
                set.delete(s[l]);
                l++;
            }
            let w = r - l + 1;
            res = Math.max(w, res);
            set.add(s[r]);
        }

        return res;
    }
}
